src/Artsy/Router/buildServerApp.tsx

```diff
@@ -105,7 +105,7 @@ export function buildServerApp(config: ServerRouterConfig): Promise<Resolve> {
             const data = await relayEnvironment.relaySSRMiddleware.getCache()
             // Render tree again, but this time with Relay data being available.
             const html = ReactDOMServer.renderToString(
```

Comment:
This was the bug fix, if you expand the lines here you'll see that `ReactDOMServer.renderToString` is called twice, and accumulates `headTags` through both passes. The bug fix here was to only ever pass those in and accumulate them once.

Reply:
I'm having trouble understanding how `headTags` is accumulating across those two calls. Is it being mutated by the `<Boot>` component? The only place I see it being set or changing is https://github.com/artsy/reaction/pull/2327/files#diff-8721a7de273dd9c3f5ca4c9838197b6eL79.

Reply:
I'm a tad fuzzy on that myself, but yes, this winds up being referenced inside `Boot`: https://github.com/artsy/reaction/blob/0d3c391bbc4233bdf96175ed144626e9f2856a5e/src/Artsy/Router/Boot.tsx#L58 

If looking at `HeadProvider`, https://github.com/tizmagik/react-head/blob/master/src/HeadProvider.js , there is indeed some `const var = this.props.headTags; var.push(...);`, etc. but I didn't trace it further.

When pairing we had a simple repro that was being popping up twice `alert("test")`, and with this fix went down to once!

Reply:
Also, if you view source, you'll see the tags all being duplicated. And with this fix the HTML looks correct (no dupes).

Reply:
It is being mutated by `Boot`, yes. 

`headTags` is an array that gets passed to the `HeadProvider`. When you use the components from `react-head`, they're actually context consumers which have a reference to that array. They basically just push stuff onto it and the ultimate rendering in the head via the portal is handled at a higher level. 

We render twice. Once for data fetching and once to actually render the view. `headTags` was being passed down both times and thus being duplicated.

Reply:
Wow, I wouldn't have expected mutation of props. I guess it makes sense for what `react-head` is doing, it's just something I totally take for granted with my strict personal "never ever mutate props" rule. (And maybe this is just another lesson to myself to never ever say never ever.)

On that note, one suggestion I'd make is that we name `headTags` something that makes it clear that it's going to be mutated, or add a comment indicating such? Definitely not required for this PR, but possibly helpful for future debuggers.